atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/sweep/queue/AbstractSweepQueueTablesTest.java

```diff
@@ -86,37 +94,70 @@ public int writeToDefault(KvsSweepQueueWriter queueWriter, long timestamp, Table
     }
 
     public int writeToCell(KvsSweepQueueWriter queueWriter, long timestamp, Cell cell, TableReference tableRef) {
+        putTimestampIntoTransactionTable(timestamp, timestamp);
+        return write(queueWriter, timestamp, cell, false, tableRef);
+    }
+
+    public int writeToCellUncommitted(KvsSweepQueueWriter queueWriter, long ts, Cell cell, TableReference tableRef) {
+        return write(queueWriter, ts, cell, false, tableRef);
+    }
+
+    public int writeToCellAborted(KvsSweepQueueWriter queueWriter, long timestamp, Cell cell, TableReference tableRef) {
+        putTimestampIntoTransactionTable(timestamp, TransactionConstants.FAILED_COMMIT_TS);
         return write(queueWriter, timestamp, cell, false, tableRef);
     }
 
     public int putTombstone(KvsSweepQueueWriter queueWriter, long timestamp, Cell cell, TableReference tableRef) {
+        putTimestampIntoTransactionTable(timestamp, timestamp);
         return write(queueWriter, timestamp, cell, true, tableRef);
     }
 
+    public int putTombstoneUncommitted(KvsSweepQueueWriter queueWriter, long ts, Cell cell, TableReference tableRef) {
+        return write(queueWriter, ts, cell, true, tableRef);
+    }
+
+    public int putTombstoneAborted(KvsSweepQueueWriter queueWriter, long ts, Cell cell, TableReference tableRef) {
+        putTimestampIntoTransactionTable(ts, TransactionConstants.FAILED_COMMIT_TS);
+        return write(queueWriter, ts, cell, true, tableRef);
+    }
+
     private int write(KvsSweepQueueWriter queueWriter, long timestamp, Cell cell, boolean isTombstone,
             TableReference tableRef) {
         WriteInfo write = WriteInfo.of(WriteReference.of(tableRef, cell, isTombstone), timestamp);
         queueWriter.enqueue(ImmutableList.of(write));
         return write.toShard(numShards);
     }
 
-    public static List<WriteInfo> writeToCellsInFixedShard(KvsSweepQueueWriter writer, long timestamp, int number,
+    protected void putTimestampIntoTransactionTable(long ts, long commitTs) {
+        try {
+            txnService.putUnlessExists(ts, commitTs);
+        } catch (KeyAlreadyExistsException e) {
+            // this is fine
```

Comment:
not if you're trying to put `commitTs = FAILED_COMMIT_TS` and the key there is a legit commit timestamp. Probably want to check and fail if that's the case.

We don't care, in this case, if we're putting some different commit ts where they're both non-failed (but by construction I don't think we'd ever do this).

Reply:
Fixed in follow-up PR